files/galaxy/tpv/tools.yml

```
@@ -583,6 +583,24 @@ tools:
       - if: input_size >= 25
         fail: Too much data, please check if the input is correct.
 
+  toolshed.g2.bx.psu.edu/repos/iuc/shovill/shovill/.*:
+    cores: 2
+    rules:
+      - if: 0.005 <= input_size < 1
+        cores: 8
+        mem: 100
+      - if: 1 <= input_size < 2
+        cores: 12
+        mem: 200
+      - if: 2 <= input_size < 20
+        cores: 20
+        mem: 350
+        scheduling:
+          prefer:
+            - condor-tpv
+      - if: input_size >= 20
+        fail: Too much data, please don't use Shovill for this
```

If shovill needs exactly the same resources as spades, then inherit from spades.

```suggestion
    inherits: toolshed.g2.bx.psu.edu/repos/nml/spades/spades/.*
```

I will suggest something for the error message so that it outputs the correct tool name.

---

The only think is that the fail error is different, so perhaps would be necessary to make it more generic.